x-pack/metricbeat/module/sql/query/query.go

```diff
@@ -104,25 +106,39 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 	return b, nil
 }
 
-// Fetch methods implements the data gathering and data conversion to the right
-// format. It publishes the event which is then forwarded to the output. In case
-// of an error set the Error field of mb.Event or simply call report.Error().
-// It calls m.fetchTableMode() or m.fetchVariableMode() depending on the response
-// format of the query.
-func (m *MetricSet) Fetch(ctx context.Context, reporter mb.ReporterV2) error {
-	db, err := sql.NewDBClient(m.Config.Driver, m.HostData().URI, m.Logger())
-	if err != nil {
-		return fmt.Errorf("could not open connection: %w", err)
+// queryDBNames returns the query to list databases present in a server
+// as per the driver name. If the given driver is not supported, queryDBNames
+// returns an empty query.
+func queryDBNames(driver string) string {
+	switch sql.SwitchDriverName(driver) {
+	case "mssql":
+		return "SELECT [name] FROM sys.databases WITH (NOLOCK) WHERE state = 0 AND HAS_DBACCESS([name]) = 1"
+		// case "mysql":
```

Comment:
Are we considering the lines commented out for future use ? If so can we add a single line comment on the top to cover why we keep these commented lines here.

Reply:
Sure.